Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/storage/dao/ContactDiaryLocationVisitDao.kt

```diff
@@ -0,0 +1,23 @@
+package de.rki.coronawarnapp.contactdiary.storage.dao
+
+import androidx.room.Dao
+import androidx.room.Query
+import de.rki.coronawarnapp.contactdiary.storage.entity.ContactDiaryLocationVisitEntity
+import kotlinx.coroutines.flow.Flow
+import org.joda.time.LocalDate
+
+@Dao
+abstract class ContactDiaryLocationVisitDao : BaseRoomDao<ContactDiaryLocationVisitEntity>() {
+
+    @Query("SELECT * FROM ContactDiaryLocationVisitEntity")
+    abstract override fun allEntries(): Flow<List<ContactDiaryLocationVisitEntity>>
+
+    @Query("SELECT * FROM ContactDiaryLocationVisitEntity WHERE id = :id")
+    abstract override suspend fun entityForId(id: Long): ContactDiaryLocationVisitEntity
+
+    @Query("SELECT * FROM ContactDiaryLocationVisitEntity WHERE date = :date")
+    abstract fun entitiesForDate(date: LocalDate): Flow<List<ContactDiaryLocationVisitEntity>>
```

Comment:
Can you explain where you think you need this? Do you plan to join Locations and LocationVisits for the "Select Contacts/Locations screen" in code?

Reply:
If you have a fragment for a specific date, you can get the relevant data very easily with this call. it avoid filtering etc.